proto/frostdb/schema/v1alpha2/schema.proto

```diff
@@ -0,0 +1,177 @@
+                        "name": "values",
+                        "storage_layout": { ... }
+                    }
+                }
+            ]
+        }
+    }
+        
+*/
+
+// Schema definition for a table.
+message Schema {
+    // Root node of the schema.
+    Group root = 2;
+    // Columns to sort by in the schema.
+    repeated SortingColumn sorting_columns = 3;
+}
+
+// Node is a node in a schema tree.
+message Node {
+    oneof type {
+        // Leaf node of the schema.
+        Leaf leaf = 1;
+        // Group of other nodes for nested schemas.
+        Group group = 2;
+    }
+}
+
+// Leaf definition.
+message Leaf {
+    // Name of the column.
+    string name = 1;
+    // Storage layout of the column.
+    StorageLayout storage_layout = 2;
+}
+
+// Group is a grouping of nodes.
+message Group {
+    // Name of the group.
+    string name = 1;
+    // Wether the group is allowed to be null.
+    bool nullable = 2;
+    // Indicates whether the group is repeated.
+    bool repeated = 3;
+    // Nodes that this group is composed of.
+    repeated Node nodes = 4;
+}
+
+// Storage layout describes the physical storage properties of a column.
+message StorageLayout {
+    // Type enum of a column.
+    enum Type {
+        // Unknown type.
+        TYPE_UNKNOWN_UNSPECIFIED = 0;
+        // Represents a string type.
+        TYPE_STRING = 1;
+        // Represents an int64 type.
+        TYPE_INT64 = 2;
+        // Represents a double type.
+        TYPE_DOUBLE = 3;
+        // Represents a boolean type.
+        TYPE_BOOL = 4;
+    }
+
+    // Type of the column.
+    Type type = 1;
+
+    // Encoding enum of a column.
+    enum Encoding {
+        // Plain encoding.
+        ENCODING_PLAIN_UNSPECIFIED = 0;
+        // Dictionary run-length encoding.
+        ENCODING_RLE_DICTIONARY = 1;
+        // Delta binary packed encoding.
+        ENCODING_DELTA_BINARY_PACKED = 2;
+        // Delta Byte Array encoding.
+        ENCODING_DELTA_BYTE_ARRAY = 3;
+        // Delta Length Byte Array encoding.
+        ENCODING_DELTA_LENGTH_BYTE_ARRAY = 4;
+    }
+
+    // Encoding of the column.
+    Encoding encoding = 2;
+
+    // Compression enum of a column.
+    enum Compression {
+        // No compression.
+        COMPRESSION_NONE_UNSPECIFIED = 0;
+        // Snappy compression.
+        COMPRESSION_SNAPPY = 1;
+        // GZIP compression.
+        COMPRESSION_GZIP = 2;
+        // Brotli compression.
+        COMPRESSION_BROTLI = 3;
+        // LZ4_RAW compression.
+        COMPRESSION_LZ4_RAW = 4;
+        // ZSTD compression.
+        COMPRESSION_ZSTD = 5;
+    }
+
+    // Compression of the column.
+    Compression compression = 3;
+
+    // Wether values in the column are allowed to be null.
+    bool nullable = 4;
+
+    // Indicates whether the parquet column is repeated.
+    bool repeated = 5;
+}
+
+// SortingColumn definition.
+message SortingColumn {
+    // Name of the column to sort by.
+    string name = 1;
```

Comment:
this can't just be a name anymore, it needs to be a "path" to the leaf, this also has consequences for our Ascending/Descending functions (eg. https://github.com/segmentio/parquet-go/blob/95f6ec7f32ee32a5e9016dcbcb0845e192e6aa41/row_group.go#L95)

Reply:
Yea, I think you're right